sdk/src/main/scala/com/horizen/account/block/AccountBlock.scala

```diff
@@ -138,6 +139,8 @@ object AccountBlock extends ScorexEncoding {
 
   def calculateTransactionsMerkleRootHash(sidechainTransactions: Seq[SidechainTypes#SCAT]): Array[Byte] = {
     // TODO: Use Eth friendly way to calculate merkle tree root.
-    Utils.ZEROS_HASH
+    // TODO: this assumes the binary representation of transactions exactly match the ethereum RLP encoding,
+    //  which is not true currently because the serializer prepends the payload with the length
+    TrieHasher.Root(sidechainTransactions.map(tx => tx.bytes).toArray)
```

Comment:
check also `AccountBlock.verifyTransactionsDataConsistency` method. Where we should verify transactions to be compatible with the root in the header